### ISSUE TITLE
Prisma duplicates large binaries 3 times in a single project

### ISSUE THREAD

**Issue by cpojer** [prisma/prisma#6878]
## Bug description
`query-engine-darwin` is 38.1 MB. In my project, I have a copy of it in `@prisma/client`, `prisma` and in a folder `node_modules/.prisma`. The two additional copies increase the size of my `node_modules` folder on disk by more than 15%, and it makes prisma by far the largest dependency of my project.

## How to reproduce

Install `@prisma/client', 'prisma' and use prisma in a project to reproduce. Any project will have these files.

## Expected behavior

Fewer copies of large binaries to reduce the size of my `node_modules` folder on disk. I suggest the creation of a `@prisma/query-engine` or `@prisma/binaries` package that will consolidate all copies of the same binaries so that only a single copy ever exists in a project.

Note: the `introspection-engine` and `migration-engine` binaries are large too. Could there be any sort of de-duplication across these binaries to further reduce the size of the `@prisma/client` package?

Thank you for your consideration :)

## Environment & setup
Latest version of prisma, on the latest version of mac OS and node.

```
prisma               : 2.21.2
@prisma/client       : 2.21.2
Current platform     : darwin
Query Engine         : query-engine e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : e421996c87d5f3c8f7eeadd502d4ad402c89464d
Studio               : 0.371.0
```

**Comment by macjuul** [comment]
Facing a similar problem here. These duplicate files are unnecessarily increasing my docker image size, so hopefully we can see a solution soon

**Comment by Axxxx0n** [comment]
Yeah, large docker sizes are making our container registry more expensive and pulls slow

**Comment by janpio** [comment]
Hey, we have not fixed the duplication problem yet but at least the files are not ~50% smaller on all platforms. ([Read more in our 4.8.0 release notes](https://github.com/prisma/prisma/releases/tag/4.8.0)). Expect more improvements in this area in the near future.
